### PR TITLE
[WIP] --fake-newly-deployed && --fake-not-newly-deployed

### PR DESCRIPTION
**Pull request by vporton**
My code does not work (the values of the options variables are always as if they were false). I don't know how to make it work.

Setting option variables happens _after_ it is used in `await deploy()` calls, this causes it not to work.

The code is also messy.

Probably, changes in Buidler core are necessary to implement this.

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -1078,6 +1080,11 @@ Plus they are only used when the contract is meant to be used as standalone when
     name: string,
     options: DeployOptions
   ): Promise<DeployResult> {
+    //log('fakeNewlyDeployed:', fakeNewlyDeployed, 'fakeNotNewlyDeployed:', fakeNotNewlyDeployed);
+    if (fakeNewlyDeployed || fakeNotNewlyDeployed) {
+      const deployment = await env.deployments.get(name);
+      return { newlyDeployed: fakeNewlyDeployed, ...deployment };
```
Comment (wighawag) [inline src/helpers.ts:1086]: I don't think this is a good approach.
newlyDeployed should always means the contract was newly deplioyed

**Comment by wighawag** [comment]
Thanks for the PR, but I am not fan of the approach. It change the meaning of the return value : `newlyDeployed`

**Comment by vporton** [comment]
With the word "fake" in the option, it is not a trouble if the value of `newlyDeployed` matches the fact of being newly deployed. It is exactly a fake. It is the right approach. Think again. Anything labeled "fake" does not need to match the data.

**Comment by wighawag** [comment]
Closing this now. If this is still something desired, let me know. there might be others way to achieve this

**Pull request closed without merging**